python/mxnet/test_utils.py

```diff
@@ -116,7 +116,22 @@ def _validate_csr_generation_inputs(num_rows, num_cols, density,
                              % (density, num_rows, num_cols))
 
 
-def _get_uniform_dataset_csr(num_rows, num_cols, density=0.1, dtype=None):
+def shuffle_csr_column_indices(csr):
+    """Shuffle CSR column indices per row
+    This allows validation of unordered column indices, which is not a requirement
+    for a valid CSR matrix
+    """
+    row_count = len(csr.indptr) - 1
+    for i in range(row_count):
+        start_index = csr.indptr[i]
+        end_index = csr.indptr[i + 1]
+        sublist = np.array(csr.indices[start_index : end_index])
+        np.random.shuffle(sublist)
+        csr.indices[start_index : end_index] = sublist
+
+
+def _get_uniform_dataset_csr(num_rows, num_cols, density=0.1, dtype=None,
```

Comment:
Oh sorry I forget to mention the context. CUSPARSE is like CUDNN, a library developed by Nvidia to speed up sparse matrix computation on GPU. http://docs.nvidia.com/cuda/cusparse/#abstract If we want to speed up CSR operators on GPU in the future, cusparse is the way to go. I asked one of their engineers and they confirmed that they only support CSR with sorted indices and no duplicates. If our format diverge from theirs, that will make integration w/ cusparse much harder